lib/ansible/modules/cloud/openstack/os_server_action.py

```diff
@@ -239,6 +247,17 @@ def main():
                 _wait(timeout, cloud, server, action, module, sdk)
             module.exit_json(changed=True)
 
+        elif action == 'reboot':
+            if not _reboot_state(status):
+                module.exit_json(changed=False)
+
+            cloud.compute.post(
+                _action_url(server.id),
+                json={'reboot': {'type': 'HARD'}})
```

Comment:
Do we really want a default and only "hard" reboot ?

Reply:
That's right. Added soft reboot as default and new argument `reboot_type`.